node/lib/DeviceInterface.ts

```diff
@@ -143,10 +146,49 @@ export class DeviceParameter extends DeviceCharacteristic {
     /**
      * JSON schema that specifies the type of the parameter.
      */
-    public readonly parameterType: Schema;
+    public readonly parameterType: JsonSchema;
 
     /**
      * True if this is an out parameter (return value), false if in.
      */
     public readonly isOut: boolean;
 }
+
+/**
+ * Describes the schema of JSON data.
+ * Reference http://json-schema.org/ and
+ * https://raw.githubusercontent.com/tdegrunt/jsonschema/master/lib/index.d.ts
+ */
+export interface JsonSchema {
+    id?: string;
```

Comment:
Yes, partly that, and also because the jsonschema typings were problematic. See the notes in the PR description. 

Someday soon, typings will be installed via NPM, so depending on them will get simpler. But as things are now I couldn't get the typings to work without also requiring any TypeScript consumer of the library to set up the jsonschema typings in a certain way.

---
In reply to: [73969187](https://github.com/openT2T/opent2t/pull/15#discussion_r73969187) [](ancestors = 73969187)